hudi-flink/src/main/java/org/apache/hudi/source/StreamReadMonitoringFunction.java

```diff
@@ -203,9 +203,11 @@ public void monitorDirAndForwardSplits(SourceContext<MergeOnReadInputSplit> cont
         instantRange = InstantRange.getInstance(specifiedStart, instantToIssue.getTimestamp(),
             InstantRange.RangeType.CLOSE_CLOSE);
       } else {
-        // first time consume and no start commit,
-        // would consume all the snapshot data PLUS incremental data set
-        instantRange = null;
+        // first time consume and no start commit. Change to consume the incremental data set
+        // of latest commit instead of all the snapshot data PLUS incremental data set.
```

Comment:
first time consume and no start commit,
consumes the latest incremental data set